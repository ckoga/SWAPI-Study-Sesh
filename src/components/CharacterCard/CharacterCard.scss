
li {
  list-style: none;
}

.character-card {
  text-align: start;
  line-height: 1.5em;
}