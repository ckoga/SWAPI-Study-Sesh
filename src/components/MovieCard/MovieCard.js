import React from 'react';
import './MovieCard.scss';
import {Link} from 'react-router-dom';



const MovieCard = ({ episode, title, releaseDate, fetchChar }) => {
  return (
    <article className='movie-card'>
      <h2>Episode {episode}</h2>
      <h3>{title}</h3>
      <p>{releaseDate}</p>
      <Link to={`/movies/${episode}`}>
        <button
          className='characters-btn'
          onClick={() => fetchChar(episode)}
        >View Characters</button>
      </Link>
    </article>
  )
};

export default MovieCard;
