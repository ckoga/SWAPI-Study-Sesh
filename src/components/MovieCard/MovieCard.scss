@import '../../variables';

.card {
  align-items: center;
  background-color: $header-background;
  color: $accent-color;
  display: flex;
  flex-direction: column;
  height: 300px;
  justify-content: space-around;
  margin: 30px;
  padding: 10px;
  text-align: center;
  width: 250px;

  &:hover {
    box-shadow: 5px 5px 5px $accent-color;
  }

  h2 {
    font-size: 2em;
    margin: 5px 0 10px 0;
  }

  h3 {
    font-size: 1.5em;
  }
}

.characters-btn {
  background-color: $header-background;
  border: 1px solid $accent-color;
  border-radius: 5px;
  box-shadow: 5px 5px 5px $background-color;
  color: $accent-color;
  cursor: pointer;
  font-size: 1em;
  margin: 0px;
  padding: 10px;

  &:hover {
    background-color: $accent-color;
    color: $header-background;
  }
}
