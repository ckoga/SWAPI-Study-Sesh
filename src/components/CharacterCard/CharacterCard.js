import React from 'react';
import '../MovieCard/MovieCard.scss';

const CharacterCard = ({ episode_id, name, homeworld, population, species, films }) => {
  return (
    <div className='card'>
      <h2>{name}</h2>
      <ul>
        <li>Homeworld: {homeworld}</li>
        <li>Population: {population}</li>
        <li>Species: {species}</li>
        <li>Films: {films}</li>
      </ul>
      <button>favorite</button>
    </div>
  )
}

export default CharacterCard;