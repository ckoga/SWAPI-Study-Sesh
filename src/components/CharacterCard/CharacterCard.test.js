import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from '../CharacterCard/CharacterCard';
import MovieCard from '../MovieCard/MovieCard';

describe('CharacterCard', () => {
  let characters;
  beforeEach(() => {
    characters = [
      {
        episode_id: 10000000000,
        name: 'Heather',
        homeworld: 'Earth',
        species: 'Human',
        films: ['The Best Movie']
      }
    ]
  })

  it('should match the snapshot with all the data being passed in correctly', () => {
    const wrapper = shallow(<CharacterCard 
      key={characters[0].name}
      name={characters[0].name}
      homeworld={characters[0].homeworld}
      species={characters[0].species}
      films={characters[0].films}    
    />)

    expect(wrapper).toMatchSnapshot()
  })
})