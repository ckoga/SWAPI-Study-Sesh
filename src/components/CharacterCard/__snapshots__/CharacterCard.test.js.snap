// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot with all the data being passed in correctly 1`] = `
<div
  className="card character-card"
>
  <h2>
    Heather
  </h2>
  <ul>
    <li>
      Homeworld: 
      Earth
    </li>
    <li>
      Population: 
    </li>
    <li>
      Species: 
      Human
    </li>
    <li>
      Films: 
      The Best Movie
    </li>
  </ul>
  <button
    className="favorite-character-btn"
  >
    favorite
  </button>
</div>
`;
